File: shared/src/commonMain/kotlin/com/colormixlab/platform/HapticProvider.kt
package com.colormixlab.platform

/**
 * Haptic feedback types supported across platforms
 */
enum class HapticType {
    LIGHT_TAP,    // Light tap/click feedback
    SUCCESS,      // Success/completion feedback
    ERROR         // Error/warning feedback
}

/**
 * Platform-independent haptic feedback interface.
 * Android uses Vibrator, iOS uses UIImpactFeedbackGenerator.
 */
expect class HapticProvider {
    /**
     * Perform haptic feedback of the specified type.
     */
    fun performHaptic(type: HapticType)
}

File: iosApp/ColorMixLab/ColorMixLab/Utilities/HapticProvider.swift
//
//  HapticProvider.swift
//  ColorMixLab iOS
//
//  Provides haptic feedback for user interactions
//

import UIKit

enum HapticType {
    case lightTap
    case success
    case error
    case warning
}

class HapticProvider {
    func performHaptic(type: HapticType) {
        switch type {
        case .lightTap:
            let generator = UIImpactFeedbackGenerator(style: .light)
            generator.impactOccurred()

        case .success:
            let generator = UINotificationFeedbackGenerator()
            generator.notificationOccurred(.success)

        case .error:
            let generator = UINotificationFeedbackGenerator()
            generator.notificationOccurred(.error)

        case .warning:
            let generator = UINotificationFeedbackGenerator()
            generator.notificationOccurred(.warning)
        }
    }
}
